iot-meter/models/account.go

```diff
@@ -5,11 +5,36 @@ import (
 )
 
 type Account struct {
-	ID         string `gorm:"primaryKey"`
+	ID         string 	`gorm:"primaryKey" gorm:"not null"`
 	Email      string
 	Password   string
-	PlanType   string
-	TotalUsers uint
+	PlanType   string 	`gorm:"default:Basic" gorm:"not null"`
+	TotalUsers uint 	`gorm:"default:0" gorm:"not null"`
 	CreatedAt  time.Time
 	UpdatedAt  time.Time
 }
+
+func (a *Account) CanAcceptMoreUsers() (bool) {
+	if a.PlanType == "Basic" && a.TotalUsers >= 100 {
+		return false
+	}
+
+	return true
+}
+
+func (a *Account) IncreaseUserCount() {
+	a.TotalUsers += 1
+}
+
+func (a *Account) UpgradePlan() (bool) {
```

Comment:
Here and everywhere else - you don't need parentheses when there's a single return value

```suggestion
func (a *Account) UpgradePlan() bool {
```